### ISSUE TITLE
Tome of Knowledge + Transmutation Chamber is no good!

### ISSUE THREAD

**Issue by MrPopannes** [pkmnfrk/equivalentintegrations#25]
As said. after using the tome of knowledge, the game was unplayable with the chamber and its use deeply integrated into my ME system.
This should be expected, but a warning might be nice. Maybe in the tooltip to not use this with the tome of knowledge.

Edit: I play PO3. i posted a similar Ticket at their github. Just so you know.

**Comment by pkmnfrk** [comment]
Well, this is very tricky. I've not played PO3, but the cause is almost certainly that it has a lot of items. But, the question is, who is the bottleneck? Is it Equivalent Integrations, Applied Energistics or ProjectE? To that end, I did a little test.

I hacked EI to register 250,000 dummy items, to simulate a very bloated mod pack. If you're wondering, in a maximized window, that's 480 pages of items in JEI, give or take.
![image](https://user-images.githubusercontent.com/1333816/53998451-c0690880-410d-11e9-8dfc-a61cfe22c5ac.png)

Sure enough, as soon as I created a very simple setup with a Storage Bus hooked up to a Transmutation Chamber, and added a Tome of Knowledge, things started to lag. With the symptoms reproduced, I started to dig.

The good news is that it's not Applied Energistics. The same thing happens without it; the lag is triggered when your EMC changes.

The less good news is that the bottle neck is inside ProjectE, not inside Equivalent Integrations. Basically, the function that reports the EMC values of items is slightly slow. Not incredibly slow, or even pretty slow, but just slow enough that calling it for the 250,000 items takes a total of about 7 seconds. As an experiment, I changed the code to not call ProjectE and just pretend everything costs 100 EMC, and it took 97ms instead of 6,936ms.

Profiling, with ProjectE:

![image](https://user-images.githubusercontent.com/1333816/53998670-8d734480-410e-11e9-8590-0cc9f1da2b13.png)

After, without ProjectE:

![image](https://user-images.githubusercontent.com/1333816/53999080-1e96eb00-4110-11e9-86e5-834611acca4e.png)

(Note that these are not directly comparable, I had to refresh the inventory about 20 times to get it to show up at all, because of how quick it is)

All that said, it is theoretically possible for me to fix this, I think. It would require me to add caching, which hopefully I could do properly. EI had caching in it before, but I determined that it was not necessary. Perhaps I will revisit this.
